manifests/service/puppetdb.pp

```diff
@@ -12,166 +12,327 @@
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'PDBReadPool_pool_Usage',
+      'mbean' => 'puppetlabs.puppetdb.database:name=PDBReadPool.pool.Usage'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'PDBReadPool_pool_Wait',
+      'mbean' => 'puppetlabs.puppetdb.database:name=PDBReadPool.pool.Wait'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'PDBWritePool_pool_ActiveConnections',
+      'mbean' => 'puppetlabs.puppetdb.database:name=PDBWritePool.pool.ActiveConnections'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'PDBWritePool_pool_IdleConnections',
+      'mbean' => 'puppetlabs.puppetdb.database:name=PDBWritePool.pool.IdleConnections'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'PDBWritePool_pool_PendingConnections',
+      'mbean' => 'puppetlabs.puppetdb.database:name=PDBWritePool.pool.PendingConnections'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'PDBWritePool_pool_TotalConnections',
+      'mbean' => 'puppetlabs.puppetdb.database:name=PDBWritePool.pool.TotalConnections'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'PDBWritePool_pool_Usage',
+      'mbean' => 'puppetlabs.puppetdb.database:name=PDBWritePool.pool.Usage'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'PDBWritePool_pool_Wait',
+      'mbean' => 'puppetlabs.puppetdb.database:name=PDBWritePool.pool.Wait'
+    }
   ]
 
   $ha_sync_metrics = [
-    { 'name' => 'ha_last-sync-succeeded',
-      'url'  => 'puppetlabs.puppetdb.ha:name=last-sync-succeeded' },
-    { 'name' => 'ha_seconds-since-last-successful-sync',
-      'url'  => 'puppetlabs.puppetdb.ha:name=seconds-since-last-successful-sync' },
-    { 'name' => 'ha_failed-request-counter',
-      'url'  => 'puppetlabs.puppetdb.ha:name=failed-request-counter' },
-    { 'name' => 'ha_sync-duration',
-      'url'  => 'puppetlabs.puppetdb.ha:name=sync-duration' },
-    { 'name' => 'ha_catalogs-sync-duration',
-      'url'  => 'puppetlabs.puppetdb.ha:name=catalogs-sync-duration' },
-    { 'name' => 'ha_reports-sync-duration',
-      'url'  => 'puppetlabs.puppetdb.ha:name=reports-sync-duration' },
-    { 'name' => 'ha_factsets-sync-duration',
-      'url'  => 'puppetlabs.puppetdb.ha:name=factsets-sync-duration' },
-    { 'name' => 'ha_nodes-sync-duration',
-      'url'  => 'puppetlabs.puppetdb.ha:name=nodes-sync-duration' },
-    { 'name' => 'ha_record-transfer-duration',
-      'url'  => 'puppetlabs.puppetdb.ha:name=record-transfer-duration' },
+    {
+      'type'  => 'read',
+      'name'  => 'ha_last-sync-succeeded',
+      'mbean' => 'puppetlabs.puppetdb.ha:name=last-sync-succeeded'
+    },
+    {
+      'type'  => 'read',
+      'name'  => 'ha_seconds-since-last-successful-sync',
+      'mbean' => 'puppetlabs.puppetdb.ha:name=seconds-since-last-successful-sync'
+    },
+    #{
+    #  'type'  => 'read',
+    #  'name'  => 'ha_failed-request-counter',
+    #  'mbean' => 'puppetlabs.puppetdb.ha:name=failed-request-counter'
+    #},
```

Comment:
I believe these aren't retrieving because the counters don't exist until a failure occurs. Looks like there's only 3, so we could sort em' out later.

Reply:
Done!